### DISCUSSION TITLE
Key authentication with one secret per service/ingress object?

### DISCUSSION THREAD

**Discussion by smartaquarius10** [Kong/kong#10600] [Help]
Hello,

How to map kong consumer for one specific service/ingress route while using kong authentication(https://docs.konghq.com/hub/kong-inc/key-auth/)

Currently, the moment we tag the kong plugin to any of the ingress route or a service they start accepting the call from any of the secret which is mapped through any of the kong consumer.. It seems to be a highly insecure approach especially in shared clusters.

Consider multiple teams created their kong consumer, secret and kong plugin in their own namespace.  Now anyone can call eachother using their own api key which is unacceptable. 

How to use one secret(as api key) per kong consumer per Ingress/Service object..

Please suggest. Thank you..

Kind Regards,
Tanul

**Comment by ghost** [comment]
I think what you need is: to restrict access to certain routes/services only to specified consumers, then you need to ACL plugin, read more: https://docs.konghq.com/hub/kong-inc/acl/
